src/assert.rs

```diff
@@ -447,6 +452,38 @@ impl OutputAssertionBuilder {
         self.assertion
     }
 
+    /// Expect the command to match **however many times** this `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    /// assert_cli::Assert::command(&["echo", "42"])
+    ///     .stdout().matches("[0-9]{2}")
+    ///     .unwrap();
+    /// ```
+    pub fn matches(mut self, output: String) -> Assert {
```

Comment:
1. Should we accept a regex?
2. Should we accept a byte slice and a byte regex?

If we decide on yes, it doesn't mean you have to do them (I don't want to bar for contributions to be perfection) but we need to at least create issues for them and keep them in mind with the design of the API / implementation.

Reply:
I think yes, at least a regex, a byte regex not so much IMHO, however, I see no problem in setting us up for later by implementing an abstraction layer here.

Reply:
Also, here the current usage code is more like `.matches(String::from("[0-9]")` as such, i will probably use str instead of string (just borrow).

Reply:
For str vs String, you can always make it [accept both](http://hermanradtke.com/2015/05/06/creating-a-rust-function-that-accepts-string-or-str.html)